core/src/main/java/org/apache/iceberg/BaseScan.java

```diff
@@ -187,7 +193,10 @@ public long targetSplitSize() {
     long tableValue =
         PropertyUtil.propertyAsLong(
             table().properties(), TableProperties.SPLIT_SIZE, TableProperties.SPLIT_SIZE_DEFAULT);
-    return PropertyUtil.propertyAsLong(context.options(), TableProperties.SPLIT_SIZE, tableValue);
+
+    long splitSize = adaptiveSplitSize(tableValue).orElse(tableValue);
```

Comment:
Shouldn't the table split size override the adaptive size? If so, we would not want to always pass in a table value.

Reply:
No, because the table size is typically going to be configured as the default.  Small tables won't align well with that.  Even if it is set to a smaller value, it may collapse too much.

Reply:
I don't quite understand. I thought most tables didn't have a split size default explicitly set and would use the Iceberg default, `TableProperties.SPLIT_SIZE_DEFAULT`. Is that not the case?

What I meant was that if the split size is explicitly set (not common, I think) then we should always use it because it is static.

Reply:
I agree with @rdblue, I thought by default we don't have this table property included in the metadata. The fields should be empty. If the property is explicitly set we should use the table value.